projects/coordinator/src/main/java/org/batfish/coordinator/WorkMgr.java

```diff
@@ -1145,44 +1188,18 @@ public void uploadTestrig(String containerName, String testrigName, InputStream
     }
     Path zipFile = CommonUtil.createTempFile("testrig", ".zip");
     CommonUtil.writeStreamToFile(fileStream, zipFile);
-    Path unzipDir = testrigDir.resolve(BfConsts.RELPATH_TEST_RIG_DIR);
+    Path unzipDir = CommonUtil.createTempDirectory("tr");
     UnzipUtility.unzip(zipFile, unzipDir);
 
-    /*-
-     * Sanity check what we got:
-     *    There should be just one top-level folder.
-     */
-    SortedSet<Path> unzipDirEntries = CommonUtil.getEntries(unzipDir);
-    if (unzipDirEntries.size() != 1 || !Files.isDirectory(unzipDirEntries.iterator().next())) {
-      CommonUtil.deleteDirectory(testrigDir);
-      throw new BatfishException(
-          "Unexpected packaging of environment. There should be just one top-level folder");
-    }
-    Path unzipSubdir = unzipDirEntries.iterator().next();
-    SortedSet<Path> subFileList = CommonUtil.getEntries(unzipSubdir);
-
-    // create empty default environment
-    Path defaultEnvironmentLeafDir =
-        testrigDir.resolve(
-            Paths.get(
-                BfConsts.RELPATH_ENVIRONMENTS_DIR,
-                BfConsts.RELPATH_DEFAULT_ENVIRONMENT_NAME,
-                BfConsts.RELPATH_ENV_DIR));
-    defaultEnvironmentLeafDir.toFile().mkdirs();
-
-    // things look ok, now make the move
-    for (Path subFile : subFileList) {
-      Path target;
-      if (isEnvFile(subFile)) {
-        target = defaultEnvironmentLeafDir.resolve(subFile.getFileName());
-      } else {
-        target = unzipDir.resolve(subFile.getFileName());
-      }
-      CommonUtil.moveByCopy(subFile, target);
+    try {
+      initTestrig(testrigDir, unzipDir, false);
+    } catch (Exception e) {
+      CommonUtil.deleteDirectory(unzipDir);
```

Comment:
Slightly simpler:

```java
try {
   init Testrig
} catch (Exception e) {
   throw BE(..., e);
} finally {
   CommonUtil.delete
   CommonUtil.delete
}
```

This way the deletion happens in both success and failure cases naturally. We also get slightly better handling of exceptions in the delete functions themselves in the failure case.

(then you don't need the ones at L1202/03)

Reply:
ah, i had forgotten that the finally will executed even if the catch block throws an exception.